sdk/keyvault/azure-security-keyvault/tests/test_key_client.py

```diff
@@ -245,55 +250,70 @@ def test_list_versions(self, vault_client, **kwargs):
 
     @ResourceGroupPreparer()
     @VaultClientPreparer(enable_soft_delete=True)
-    def test_recover_purge(self, vault_client, **kwargs):
-
+    def test_recover(self, vault_client, **kwargs):
         self.assertIsNotNone(vault_client)
         client = vault_client.keys
         keys = {}
 
         # create keys to recover
         for i in range(self.list_test_size):
-            key_name = self.get_resource_name("keyrec{}".format(i))
-            keys[key_name] = client.create_key(key_name, "RSA")
-
-        # create keys to purge
-        for i in range(self.list_test_size):
-            key_name = self.get_resource_name("keyprg{}".format(i))
+            key_name = "key{}".format(i)
             keys[key_name] = client.create_key(key_name, "RSA")
 
         # delete all keys
         for key_name in keys.keys():
             client.delete_key(key_name)
 
-        if self.is_live:
-            time.sleep(20)
+        for key_name in keys.keys():
+            self._poll_until_no_exception(
+                functools.partial(client.get_deleted_key, key_name), expected_exception=ResourceNotFoundError
+            )
 
         # validate all our deleted keys are returned by list_deleted_keys
         deleted = [s.name for s in client.list_deleted_keys()]
-
         self.assertTrue(all(s in deleted for s in keys.keys()))
 
         # recover select keys
-        for key_name in [s for s in keys.keys() if s.startswith("keyrec")]:
+        keys_to_recover = [s for s in keys.keys() if s.startswith("keyrec")]
```

Comment:
key_name no longer starts with `keyrec`?
would `keys_to_recover` have any keys at all?

Reply:
It would not. Good eye, thanks! :mag: